rskj-core/src/main/java/co/rsk/peg/bitcoin/BitcoinUtils.java

```diff
@@ -179,11 +178,28 @@ public static void addSpendingFederationBaseScript(BtcTransaction btcTx, int inp
     }
 
     private static void setSegwitScriptSig(TransactionInput txIn, Script redeemScript) {
+        Script segwitScriptSig = buildSegwitScriptSig(redeemScript);
+        txIn.setScriptSig(segwitScriptSig);
+    }
```

Comment:
I think you don't need this method.  You should call `buildSegwitScriptSig` from `addSpendingFederationBaseScript`. Imo, this method does not add any value.

Reply:
Updated.